elasticsearch/store/_utilities.py

```diff
@@ -0,0 +1,153 @@
+from typing import Any, Dict, List, Optional, Union
+
+import numpy as np
+from elasticsearch import (
+    AsyncElasticsearch,
+    BadRequestError,
+    ConflictError,
+    NotFoundError,
+)
+
+Matrix = Union[List[List[float]], List[np.ndarray], np.ndarray]
+
+
+def create_elasticsearch_client(
+    agent_header: str,
+    client: Optional[AsyncElasticsearch] = None,
+    url: Optional[str] = None,
+    cloud_id: Optional[str] = None,
+    api_key: Optional[str] = None,
+    username: Optional[str] = None,
+    password: Optional[str] = None,
+    client_params: Optional[Dict[str, Any]] = None,
```

Comment:
I moved this function to the orchestration to not have to repeat the silly ifs. We can totally change the names if we keep this function here. I'm also not opposed to duplicating this if you feel it's not a good native fit for the client lib. 

(Currently users can specify their details as arguments to the the 3rd party libs vector stores. They already work with some of these names so on that end we are bound to them.)

Reply:
`agent_header` is new and not used anywhere yet, will rename to `user_agent`.

Reply:
> I moved this function to the orchestration to not have to repeat the silly ifs. We can totally change the names if we keep this function here. I'm also not opposed to duplicating this if you feel it's not a good native fit for the client lib.

Yeah I'm not sure to be honest. The native way to do that would be to just accept a client instance. I'm wondering if we should stop calling `create_elasticsearch_client` from `ElasticsearchEmbeddings.__init__` and `ElasticsearchStore.__init__` but accept a client and just call `.options` there. Then in the third-party libraries we can call `create_elastiscearch_client` independently. This is more friction though.

> (Currently users can specify their details as arguments to the the 3rd party libs vector stores. They already work with some of these names so on that end we are bound to them.)

We're not bound to them here just yet, though: the translation from say `username`/`password` to `basic_auth` could happen in the 3rd party libs vector stores.

Reply:
I agree with @pquentin to design the code to be more client native. I think it's better to pass just a client instance, instead of all the possible parameters for building a client. In the 3rd party integrations we can then decide to pass the parameter and build a client object to be passed to the client helper function.